backend/src/graphql_api/search_result.rs

```diff
@@ -673,6 +675,134 @@ impl SearchResult {
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(
+            desc = "Returns the elements in the list that come before the specified cursor."
+        )]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, PltToken>> {
+        let pool = get_pool(ctx)?;
+        let config = get_config(ctx)?;
+        let query = ConnectionQuery::<DescendingI64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.plt_token_events_collection_limit,
+        )?;
+        let mut connection = connection::Connection::new(false, false);
+
+        // Allow alphanumeric search for token names and token IDs
+        // This regex allows letters, numbers, spaces, and common symbols for token names/symbols
+        let plt_search_regex: Regex = Regex::new(r"^[a-zA-Z0-9\s\-_\.]+$")
+            .map_err(|_| InternalError::InternalError("Invalid regex".to_string()))?;
+
+        if self.query.trim().is_empty() || self.query.len() < 2 {
+            return Ok(connection);
+        }
+
+        // Check if it could be a Base58 address (for issuer search)
+        let base58_regex: Regex = Regex::new(r"^[1-9A-HJ-NP-Za-km-z]+$")
+            .map_err(|_| InternalError::InternalError("Invalid regex".to_string()))?;
+
+        let is_base58 = base58_regex.is_match(&self.query);
+        let is_general_search = plt_search_regex.is_match(&self.query);
+
+        if !is_base58 && !is_general_search {
+            return Ok(connection);
+        }
+
+        let lower_case_query = self.query.to_lowercase();
+
+        // First, get the indices of matching PLT tokens
+        let index_rows = sqlx::query!(
+            r#"SELECT plt_tokens.index
+            FROM plt_tokens
+            LEFT JOIN accounts ON plt_tokens.issuer_index = accounts.index
+            WHERE (
+                -- Search by token name (case insensitive)
+                LOWER(plt_tokens.name) LIKE $5 || '%'
+                -- Search by token ID (case insensitive) 
+                OR LOWER(plt_tokens.token_id) LIKE $5 || '%'
+                -- Search by issuer address if it's a valid Base58 format
+                OR ($6 AND accounts.address LIKE $7 || '%')
+            )
+            AND $2 < plt_tokens.index 
+            AND plt_tokens.index < $1
+            ORDER BY
+                (CASE WHEN $4 THEN plt_tokens.index END) ASC,
+                (CASE WHEN NOT $4 THEN plt_tokens.index END) DESC
+            LIMIT $3"#,
+            i64::from(query.from), // $1
+            i64::from(query.to),   // $2
+            query.limit,           // $3
+            query.is_last,         // $4
+            lower_case_query,      // $5 - for name and token_id search
+            is_base58,             // $6 - flag to enable address search
```

Comment:
Why was the toggle for this query introduced? Is it for optimization as in most cases (when we don't have a `base58` user input), that part of the query can be skipped?

Would be good to leave a comment why it is there.